### PR TITLE
chore: 3.3.x compilation fixes

### PR DESCRIPTION
**Pull request by merlin-northern**
3.3.x is failing: https://gitlab.com/Northern.tech/Mender/mender/-/pipelines/943410534

```
# github.com/mendersoftware/mender/app
app/auth.go:266:15: undefined: timers
app/auth.go:279:5: undefined: timers
app/auth.go:295:15: undefined: timers
app/auth.go:304:5: undefined: timers
app/auth.go:413:16: undefined: timers
app/auth.go:418:6: undefined: timers
app/auth.go:476:13: undefined: timers
app/auth.go:481:3: undefined: timers
make: *** [Makefile:85: mender] Error 1
```

Ticket: None

### CODE REVIEW

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline please honey

File: app/state.go [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-//    Unless required by applicable law or agreed to in writing, software
-//    distributed under the License is distributed on an "AS IS" BASIS,
-//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-//    See the License for the specific language governing permissions and
-//    limitations under the License.
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
```
Comment (kacf) [inline app/state.go:13]: Please include in the commit message where you cherry-picked this from, even if it's not the exact same patch content (even more important then, actually).

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -50,6 +50,8 @@ test:modules-artifact-gen:
     - echo "deb [arch=$(dpkg --print-architecture)] https://downloads.mender.io/repos/debian ubuntu/jammy/experimental main" | tee /etc/apt/sources.list.d/mender.list
     - apt update && apt install -yy mender-artifact
     # Test dependencies
+    - pip uninstall pytest || true
+    - pip install pytest
```
Comment (kacf) [inline .gitlab-ci.yml:54]: Why?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ stages:
   - trigger
 
 variables:
-  GOLANG_VERSION: "1.17.6"
+  GOLANG_VERSION: "1.17.13"
```
Comment (kacf) [inline .gitlab-ci.yml:8]: I won't review this commit until it has a more descriptive message. 🙂

**Review by kacf — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (merlin-northern) [reply .gitlab-ci.yml:54]: `TypeError: required field "lineno" missing from alias` totally out of the blue

File: app/state.go (hunk 1 above)
Comment (merlin-northern) [reply app/state.go:13]: [fixed](https://github.com/mendersoftware/mender/pull/1316/commits/d29311f541d13253719d04e348e580a89f7c0620).

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -42,23 +42,26 @@ test:
 
 test:modules-artifact-gen:
   stage: test
-  image: ubuntu:22.04
+  image: ubuntu:18.04
```
Comment (kacf) [inline .gitlab-ci.yml:45]: I assume that this is because of OpenSSL, so please write that in the commit message.

Comment (merlin-northern) [reply .gitlab-ci.yml:45]: python3.9 apparently this is the only way. [fixed](https://github.com/mendersoftware/mender/pull/1316/commits/c8988152f7deb4730c35b6aeea6a7b23f9691897).

Comment (kacf) [reply .gitlab-ci.yml:45]: Aha! It's good that you added it then, since I misunderstood it!

**Review by kacf — APPROVED** [review]

**Comment by merlin-northern** [comment]
I am merging this one, another fix is needed for integration, we worked on it with Roberto.

File: .gitlab-ci.yml (hunk 4 above)
Comment (merlin-northern) [reply .gitlab-ci.yml:45]: I never fully understood it, so we are even :)

**Pull request merged**